### DISCUSSION TITLE
After removing the background, the text color changes

### DISCUSSION THREAD

**Discussion by jueqing1015** [dlemstra/Magick.NET#1369] [Help]
Form1.PicImage.ColorFuzz = new ImageMagick.Percentage(20));
            Form1.PicImage.Settings.FillColor = Form1.PicImage.BackgroundColor;
            Form1.PicImage.Opaque(Form1.PicImage.BackgroundColor, MagickColors.White);
The final goal is to change the color of the background to white and deepen the color of the text
![test](https://user-images.githubusercontent.com/65677359/229060033-2065132b-f1b0-4c1a-9def-ec8219d5fcf2.jpg)
![002](https://user-images.githubusercontent.com/65677359/229060834-1dfd727b-23bb-4792-a800-e8f3762c641f.jpg)

**Comment by dlemstra** [comment]
I don't understand what you are asking or telling me?

**Reply by jueqing1015** [reply]
@dlemstra 
I wanted to remove the background from the image, but using the above method, I found that the color of the text changed after removing the background.

**Reply by dlemstra** [reply]
Is it possible that you are converting from a format that has transparency (e.g. tiff/png) to another format that doesn't have transparency (e.g jpeg)?

**Reply by jueqing1015** [reply]
No, I used RGB and jpg images, and now I want to replace the background with white, but after using the above code operation, the entire text color has changed. I don't know where the problem is

**Reply by dlemstra** [reply]
I suspect that you are doing something extra that changes the colors? With your sample code I am getting a completly different output image:

```C#
using var image = new MagickImage("input.jpg");
image.ColorFuzz = new Percentage(40); // had to increase this to get a similar image.
image.Settings.FillColor = image.BackgroundColor;
image.Opaque(image.BackgroundColor, MagickColors.White);
image.Write("output.jpg");
```

**Reply by jueqing1015** [reply]
Can you upload your output

**Reply by jueqing1015** [reply]
I tested using the code you provided, but the text color obviously changed.
![image](https://user-images.githubusercontent.com/65677359/229295091-8b1320f9-d35a-40fc-90b6-6e80ecde5c8a.png)

![image](https://user-images.githubusercontent.com/65677359/229295052-833edd0e-bbf5-4bc7-847d-7be014f31c08.png)

**Reply by dlemstra** [reply]
The text color hasn't changed as much as you think it did. Due to to new background color the output looks different. And because you are saving your output image as a jpeg file the color while slightly change. When you examine the pixel at `332x263` you will see that the value changed from `#2A3A76` to `#293975`. This really is a minor change. If you want to make sure you get the exact same color I would advise you to save the image as a `png` file instead.

**Reply by jueqing1015** [reply]
Thank you very much for your answer. I will try to save it as PNG in code to test it a bit
